### PR TITLE
Notif testing

### PR DESCRIPTION
**Pull request by anubhavitis**

### CODE REVIEW

File: src/interface/cron/events/addedCollaborator.js [hunk 1]
```diff
@@ -26,37 +27,33 @@ agenda.define(jobs.ADDED_COLLABORATOR, async (job, done) => {
...
-      getLastAddedCollaboratorCheckpoint({ addedCollaborators, batchSize, latestBlockNumber });
-    if (lastAddedCollaboratorCheckpoint) {
-      await updateAddedCollaboratorCheckpoint(lastAddedCollaboratorCheckpoint);
-    }
     done();
   } catch (err) {
     console.error("Error in job", jobs.ADDED_COLLABORATOR, err);
     done(err);
   } finally {
+    if (addedCollaborators.length > 0) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedCollaborator.js:35]: This should be done before done is called in try block. Also lets not update the original logic

File: src/interface/cron/events/addedCollaborator.js [hunk 2]
```diff
@@ -26,37 +27,33 @@ agenda.define(jobs.ADDED_COLLABORATOR, async (job, done) => {
...
-  return parseInt(firstObject?.latestBlock?.number, 10) || 0;
-}
-
 async function fetchAddedCollaboratorCheckpoint() {
   const eventProcessed = await EventProcessor.findOne({});
   return eventProcessed ? eventProcessed.addedCollaborator : 0;
 }
 
 async function fetchAddedCollaboratorEvents(checkpoint, itemCount) {
+  const fetchedEvents = await fetchAddedEventsID(EVENT_NAME);
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedCollaborator.js:48]: rename to `existingEventIds`

File: src/interface/cron/events/addedCollaborator.js [hunk 3]
```diff
@@ -26,37 +27,33 @@ agenda.define(jobs.ADDED_COLLABORATOR, async (job, done) => {
...
   const response = await axios.post(API_URL, {
     query: `{
-      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, where: { blockNumber_gte : ${checkpoint} }) {
+      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, 
+        where: {
+          blockNumber_gte : ${checkpoint},
+          id_not_in:[${fetchedEvents.map(event => `"${event}"`).join(', ')}]
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedCollaborator.js:54]: This should also work `id_not_in: fetchedEvents`?

File: src/interface/cron/events/addedFiles.js [hunk 4]
```diff
@@ -1,63 +1,55 @@
...
-    });
-    if (lastAddedFilesCheckpoint) {
-      await updateAddedFilesCheckpoint(lastAddedFilesCheckpoint);
-    }
-    done();
   } catch (err) {
     console.error("Error in job", jobs.ADDED_FILE, err);
     done(err);
   } finally {
+    if (addedFiles.length > 0) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedFiles.js:30]: Same as above

File: src/interface/cron/events/addedFiles.js [hunk 5]
```diff
@@ -1,63 +1,55 @@
...
   const response = await axios.post(API_URL, {
     query: `{
-      ${EVENT_NAME}(first: ${
-      itemCount || 5
-    }, orderDirection: asc, orderBy: blockNumber, where: { blockNumber_gte : ${checkpoint} }) {
+      ${EVENT_NAME}(first: ${itemCount || 5
+      }, orderDirection: asc, orderBy: blockNumber, 
+      where: {
+        blockNumber_gte : ${checkpoint},
+        id_not_in:[${fetchedEvents.map(event => `"${event}"`).join(', ')}]
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedFiles.js:51]: same as above

File: src/interface/cron/events/addedFiles.js [hunk 6]
```diff
@@ -1,63 +1,55 @@
...
-  return parseInt(firstObject?.latestBlock?.number, 10) || 0;
-}
-
 async function fetchAddedFilesCheckpoint() {
   const eventProcessed = await EventProcessor.findOne({});
   return eventProcessed ? eventProcessed.addedFiles : 0;
 }
 
 async function fetchAddedFilesEvents(checkpoint, itemCount) {
+  const fetchedEvents = await fetchAddedEventsID(EVENT_NAME);
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/addedFiles.js:44]: same as above

File: src/interface/cron/events/editedFiles.js [hunk 7]
```diff
@@ -1,63 +1,59 @@
...
-  return parseInt(firstObject?.latestBlock?.number, 10) || 0;
-}
 
 async function fetchEditedFilesCheckpoint() {
   const eventProcessed = await EventProcessor.findOne({});
   return eventProcessed ? eventProcessed.editedFiles : 0;
 }
 
 async function fetchEditedFilesEvents(checkpoint, itemCount) {
+  const fetchedEvents = await fetchAddedEventsID(EVENT_NAME);
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/editedFiles.js:49]: same comment

File: src/interface/cron/events/editedFiles.js [hunk 8]
```diff
@@ -1,63 +1,59 @@
...
   const response = await axios.post(API_URL, {
     query: `{
-      ${EVENT_NAME}(first: ${
-      itemCount || 5
-    }, orderDirection: asc, orderBy: blockNumber, where: { blockNumber_gte : ${checkpoint} }) {
+      ${EVENT_NAME}(first: ${itemCount || 5
+      }, orderDirection: asc, orderBy: blockNumber, where: { 
+        blockNumber_gte : ${checkpoint},
+        id_not_in:[${fetchedEvents.map(event => `"${event}"`).join(', ')}]
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/editedFiles.js:55]: same as above

File: src/interface/cron/events/mint.js [hunk 9]
```diff
@@ -1,55 +1,51 @@
...
-    const lastMintCheckpoint = getLastMintCheckpoint({ mints, batchSize, latestBlockNumber });
-    if (lastMintCheckpoint) {
-      await updateMintCheckpoint(lastMintCheckpoint);
-    }
     done();
   } catch (err) {
     console.error("Error in job", jobs.MINT, err);
     done(err);
   } finally {
+    if (mints.length > 0) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/mint.js:28]: Lets not update the original logic here same comment as above

File: src/interface/cron/events/registeredCollaboratorKey.js [hunk 10]
```diff
@@ -27,39 +28,34 @@ agenda.define(jobs.REGISTERED_COLLABORATOR_KEY, async (job, done) => {
...
-      await updateRegisteredCollaboratorKeyCheckpoint(
-        lastRegisteredCollaboratorKeyCheckpoint
-      );
-    }
     done();
   } catch (err) {
     console.error("Error in job", jobs.REGISTERED_COLLABORATOR_KEY, err);
     done(err);
   } finally {
+    if (registeredCollaboratorKey.length > 0) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/registeredCollaboratorKey.js:36]: Same as above comment

File: src/interface/cron/events/removedCollaborator.js [hunk 11]
```diff
@@ -25,39 +27,35 @@ agenda.define(jobs.REMOVED_COLLABORATOR, async (job, done) => {
...
-        lastRemovedCollaboratorCheckpoint
-      );
-    }
+
     done();
   } catch (err) {
     console.error("Error in job", jobs.REMOVED_COLLABORATOR, err);
     done(err);
   } finally {
+    if (removedCollaborators.length > 0) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/removedCollaborator.js:36]: Same as above

File: src/interface/cron/events/removedCollaborator.js [hunk 12]
```diff
@@ -25,39 +27,35 @@ agenda.define(jobs.REMOVED_COLLABORATOR, async (job, done) => {
...
   return eventProcessed ? eventProcessed.removedCollaborator : 0;
 }
 
 async function fetchRemovedCollaboratorEvents(checkpoint, itemCount) {
+  const fetchedEvents = await fetchAddedEventsID(EVENT_NAME);
   const response = await axios.post(API_URL, {
     query: `{
-      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, where: { blockNumber_gte : ${checkpoint} }) {
+      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, 
+        where: {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/removedCollaborator.js:55]: Please update at all places same comment

**Review by nadeem-fileverse — CHANGES_REQUESTED** [review]
PTAL

File: src/interface/cron/events/addedCollaborator.js (hunk 3 above)
Comment (anubhavitis) [reply src/interface/cron/events/addedCollaborator.js:54]: nope. Got error with that. 
query needs the list in exact list format. 
`id_not_in: [ "abc, "bcd" ]`

Comment (nadeem-fileverse) [reply src/interface/cron/events/addedCollaborator.js:54]: Can you paste an example fetchedEvents array here .. it should be in this format only right

Comment (anubhavitis) [reply src/interface/cron/events/addedCollaborator.js]: image

File: src/interface/cron/events/process.js [hunk 13]
```diff
@@ -2,15 +2,18 @@ const { EventProcessor, Event } = require("../../../infra/database/models");
...
+const FetchEventCount = constants.CRON.PROCESS_LIMIT;
 
 agenda.define(jobs.PROCESS, async (job, done) => {
   try {
-    const minCheckpoint = await fetchMinCheckpoint();
-    const events = await fetchEvents(minCheckpoint, FetchEventCount);
-    console.log("Received entries", jobs.PROCESS, events.length);
-    await processStoredEvents(events);
+    const checkpoints = await fetchCheckspoints();
+    for (const checkpoint of checkpoints) {
```
Comment (nadeem-fileverse) [inline src/interface/cron/events/process.js:12]: Wont this make redundant requests to db in case some checkpoints are already processed ?

**Review by nadeem-fileverse — APPROVED** [review]
LGTM

**Pull request merged**